packages/dds/tree/src/simple-tree/api/schemaFactory.ts

```diff
@@ -1298,3 +1307,5 @@ export function structuralName<const T extends string>(
 	}
 	return `${collectionName}<${inner}>`;
 }
+
+const globalIdentifierAllocator: IIdCompressor = createIdCompressor();
```

Comment:
nit: 
```suggestion
const globalIdentifierCompressor: IIdCompressor = createIdCompressor();
```

Reply:
This is never used to compress anything, and is only used to allocate identifiers, so I think its current name is more accurate and descriptive.

This was moved from objectNode btw, it's not new here. I have now added a doc comment to help clarify.

Reply:
Thanks. I was coming at this more from a discoverability standpoint - e.g. "Where is that global compressor thing again? I'll try symbol-searching for 'globalCompressor'" and it'd be harder to find. But I agree that allocator here is a more apt name. This is fine. Maybe we should actually have a `type IdentifierAllocator = Pick<IIdCompressor, "generateId">` to make it very clear *shrug*.